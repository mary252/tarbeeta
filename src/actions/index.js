export * from "./departments";
export * from "./user";
export * from "./global";
export * from "./notification";
export * from "./addtocartpopup";
