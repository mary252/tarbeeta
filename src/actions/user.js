import {
  fetchAddress,
  addAddress,
  fetchInfo,
  removeAddress
} from "../services/UserService";
import {
  FETCH_ADDRESS,
  SAVE_ADDRESS,
  DELETE_ADDRESS,
  UPDATE_ADDRESS,
  STORE_USER_INFO
} from "./types";

export function store_user_info(info) {
  return {
    type: "STORE_USER_INFO",
    payload: info
  };
}

export function toggle_view() {
  return {
    type: "TOGGLE_VIEW",
    payload: null
  };
}

export function set_view_mode(mode) {
  return {
    type: "SET_VIEW_MODE",
    payload: mode
  };
}

export function set_locale(locale) {
  return {
    type: "SET_LOCALE",
    payload: locale
  };
}

export function init_notification(options) {
  options["visible"] = true;

  return async dispatch => {
    try {
      dispatch({
        type: "INIT_NOTIFICATION",
        payload: options
      });

      let t = setTimeout(() => {
        dispatch(
          {
            type: "HIDE_NOTIFICATION",
            payload: options
          },
          () => {
            clearInterval(t);
          }
        );
      }, 3000);
    } catch (e) {}
  };
}

export function loadInfo(lang_Id) {
  return async dispatch => {
    try {
      let res = await fetchInfo(lang_Id);

      let resWithReversedAddresses = {...res.data, ...res.data.address.reverse()}
 
      let payload = resWithReversedAddresses;

      dispatch({
        type: STORE_USER_INFO,
        payload
      });

      // localStorage.setItem("address_id", res.data.address[0].id);
    } catch (e) {}
  };
}

export function getAddress(langId) {
  
  return async dispatch => {
    try {
      let res = await fetchAddress(langId);
      dispatch({
        type: FETCH_ADDRESS,
        payload: res.data
      });
    } catch (e) {}
  };
}

export function deleteAddress(addressId) {
  return async dispatch => {
    try {
      let res = await removeAddress(addressId);

      dispatch({
        type: DELETE_ADDRESS,
        payload: addressId
      });
    } catch (e) {
    }
  };
}

export function addNewAddress(payload) {
  let reverseAddress = payload.reverse();
  localStorage.setItem("address_id", payload[0].id);
  return {
    type: UPDATE_ADDRESS,
    payload: reverseAddress
  };
}
