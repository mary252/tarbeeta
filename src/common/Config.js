// export const baseURL = "https://api.tarbeeta.com/v1/";
// export const v2_url = "https://api.tarbeeta.com/v2/";
export const v2_url = "https://staging.tarbeeta.com/v2/";
export const baseURL = "https://staging.api.tarbeeta.com/v1/";
export const socketUrl = "https://socket.tarbeeta.com:8878";
// export const baseURL = "http://192.168.1.15:3000/v1/";
// export const baseURL = "http://197.167.99.223:3126/v1/";
 //export const baseURL = "http://msi-laptop:3000/v1/";
export const assetUrl = "https://static.tarbeeta.com/images/";
export const baseUrlStatic = "https://www.tarbeeta.com";

//export const google_capcha="6Lf34qkUAAAAAKBh0Qs_9OZrBeZhikwqF6L7G62p"
// export const google_capcha="6LfE5akUAAAAAHC3KCKqJh1NGup58xJwQbjZxYgg"
export const google_capcha = "6LfXz6oUAAAAABbSRcTAPsK971UiPQZ6CpSSPWr4";
export const EN = 1;
export const AR = 2;
export const EG = 1;
export const DEFAULT_LOCALE = "ar";

// Facebook
export const FACEBOOK_APP_ID = "520130485093535";
export const FACEBOOK_PAGE_ID = "1996928467010956";
export const FACEBOOK_CHAT_POPUP_THEME_COLOR = "#5a9ae5";

export const PRODUCT_SLIDER_LIMIT = 18;

export const ORDER_STATUS_FAILED = 1;
export const ORDER_STATUS_PLACED = 2;
export const ORDER_STATUS_PAID = 3;
export const ORDER_STATUS_PENDING_PAYMENT = 4;
export const ORDER_STATUS_PENDING_SHIPPING = 5;
export const ORDER_STATUS_ON_THE_WAY = 6;
export const ORDER_STATUS_DELIVERED = 7;
export const ORDER_STATUS_CANCELLED = 8;
export const ORDER_STATUS_PENDING = 9;

export const APPROVED_PRODUCT = 4;

export const CASH_ON_DELEVERY = 4;
export const PAY_WALLET = 3;
export const MASTER_CARD = 1;
export const KIOSK = 2;

export const SORT_MOST_RECENT = 1;
export const SORT_OLD_TO_NEW = 2;
export const SORT_HIGHEST_PRICE = 3;
export const SORT_LOWEST_PRICE = 4;
export const SORT_RATING = 5;
export const CARD_PAYMENT_IFRAME = `https://accept.paymobsolutions.com/api/acceptance/iframes/`;
export const EN_FRAME_ID = 7340;
export const AR_FRAME_ID = 7459;
export const pending_product = 1;
