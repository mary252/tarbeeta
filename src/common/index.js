export * from "./Images";
export * from "./Config";
