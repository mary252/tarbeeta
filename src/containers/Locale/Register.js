import React, { Component } from "react";
import "./index.css"

class Register extends Component{
    render(){
        return(
            <div className="container">
                <div className="columns is-mobile">
                    <div className="column is-1"/>
                    <div className="column is-10">
                        <div className="inner-page">
                             <div className="columns">
                                <div className="column is-2"/>
                                <div className=" column is-8">

                                </div>
                             </div>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}
export default Register