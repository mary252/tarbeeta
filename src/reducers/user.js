import {
  FETCH_ADDRESS,
  UPDATE_ADDRESS,
  STORE_USER_INFO,
  DELETE_ADDRESS
} from "../actions/types";
import { updateObject } from "../ultils";

const INITIAL_STATE = {
  view_mode: "buyer",
  firstname: null,
  lastname: null,
  avatar: "",
  cover: null,
  bio: null,
  overall_rating: 4,
  n_reviews: 325,
  n_followers: 1322,
  locale: null,
  verified: false,
  mobile_verified: false,
  address: [],
  notification: [],
  wallet: null,
  mobile: null,
  shop_username: null,
  shop_id: null,
  me_busy: false,
  is_fetching_finsihed:false
};

export default function reducer(state = INITIAL_STATE, action) {
  switch (action.type) {
    case STORE_USER_INFO:
      state = updateObject(state, action.payload);
      break;

    case "TOGGLE_VIEW":
      state = {
        ...state,
        view_mode: state.view_mode === "seller" ? "buyer" : "seller"
      };
      break;

    case "INIT_NOTIFICATION":
      let old = state.notification.slice();
      old.push(action.payload);

      state = {
        ...state,
        notification: old
      };

      break;

    case "HIDE_NOTIFICATION":
      old = state.notification.splice(1, state.notification.length - 1);

      state = {
        ...state,
        // notification: state.notification.splice(state.notification, 1)
        notification: old
      };

    case "SET_VIEW_MODE":
      state = {
        ...state,
        view_mode: action.payload
      };
      break;

    case "SET_LOCALE":
      state = {
        ...state,
        locale: action.payload
      };
      break;
    case "FETCH_ADDRESS":
      state = updateObject(state, { address: action.payload });
      break;
    case DELETE_ADDRESS:
      state = updateObject(state, {
        address: state.address.filter(x => x.id !== action.payload)
      });
      break;

    case UPDATE_ADDRESS:
      state = updateObject(state, { address: action.payload });
      break;
    

      
    default:
  }
  return state;
}
