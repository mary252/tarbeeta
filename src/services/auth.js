import request from "../ultils/ApiCentral";
import { baseURL, EN, AR } from "../common";
import { GistToMember } from "./GistService";
var popupTools = require("../ultils/PopupTools");
// import { store_user_info } from "../actions";

// import store from "../Store";

export function forgetPass(data) {
  return request({
    url: `/user/forget_password`,
    method: "POST",
    data
  });
}

export function cahngePass(data) {
  return request(
    {
      url: `/user/change_password`,
      method: "POST",
      data
    },
    true
  );
}

export function setPass(data) {
  return request({
    url: `/user/reset_password`,
    method: "POST",
    data
  });
}
export function Quick_Register (data){
  return request({
    url:  "/user/quick",
    method: "POST",
    data
  });
}
export function sendVerificationEmail() {
  return request(
    {
      url: `/user/mail/send`,
      method: "POST"
    },
    true
  );
}

export function sendMobileConfirmationCode() {
  return request(
    {
      url: `/user/mobile/send`,
      method: "POST"
    },
    true
  );
}

export function verifyEmail(data, lang) {
  return request(
    {
      url: `/user/verify?lang_id=${lang}`,
      method: "POST",
      data
    },
    true
  );
}

export function verifyMobile(data, lang) {
  return request(
    {
      url: `/user/mobile/verify?lang_id=${lang}`,
      method: "POST",
      data
    },
    true
  );
}

export function validateResetCode(data) {
  return request({
    url: `/user/validate_reset_code`,
    method: "POST",
    data
  });
}

export let doAfterLogin = async function(err, data, lang) {
  if (err || !data || !data.token) {
    return;
  }

  const { token, data: { mobile_verified } = {} } = data;

  localStorage.setItem("access_token", token);
  localStorage.setItem("view_mode", "buyer");

  if (!mobile_verified) {
    localStorage.setItem("destination", "mobile");
  }

  await GistToMember(lang, data.token);
};

export function loginFacebook(lang) {
  let lang_id = lang == "en" ? EN : AR;

  popupTools.popup(
    `${baseURL}fb_login?lang_id=${lang_id}`,
    "Facebook Login",
    { width: 600, height: 600 },
    (err, data) => doAfterLogin(err, data, lang),
    true
  );
}
