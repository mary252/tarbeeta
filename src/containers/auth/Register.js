import React, { Component } from "react";
import "./css/EN.css";
import axios from "axios";
import { Redirect } from "react-router-dom";
import { TextInput, TButton, DropDown, Page } from "../../components";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faFacebookSquare } from "@fortawesome/free-brands-svg-icons";
import { loginFacebook, GistToMember } from "../../services";
import { connect } from "react-redux";
import * as user from "../../actions/user";
import { validate } from "../../ultils/validate/validation";
import { baseURL } from "../../common";
import {fetchUUid} from "../../ultils"

export class Register extends Component {
  state = {
    error_msg: null,
    firstname: "",
    country_code: "+2",
    lastname: "",
    emailRegister: "",
    mobile: "",
    password: "",
    is_loading: false,
    loggedin: false,
    firstnameError: "",
    lastnameError: "",
    mobileError: "",
    passwordError: "",
    emailRegisterError: "",
    verify_robot: false
  };

  componentWillMount = () => {
    if (localStorage.getItem("access_token")) {
      this.setState({
        loggedin: true
      });
    }
  };

  store_data = (event, input) => {
    this.setState({
      [input]: event.target.value
    });
  };

  validation = fieldsArray => {
    this.setState({
      error_msg: null
    });

    return new Promise(async (resolve, reject) => {
      const errorsArray = await fieldsArray.map((field, i) => {
        return { [`${field}Error`]: validate(field, this.state[field]) };
      });

      let hasErrors = false;

      errorsArray.map((error, j) => {
        let key = Object.keys(error)[0],
          value = Object.values(error)[0];

        this.setState({
          [key]: value
        });

        if (value !== null) {
          hasErrors = true;
        }
      });

      if (hasErrors) {
        reject(false);
      }

      resolve(true);
    });
  };

  getFieldsToValidate = () => {
    let result = ["firstname", "lastname", "mobile", "password"];
    // if (this.state.password) result.push("password");

    if (this.state.emailRegister) result.push("emailRegister");

    return result;
  };

  register = () => {
    // if (this.state.verify_robot) {
    this.validation(this.getFieldsToValidate())
      .then(() => {
        this.setState({
          is_loading: true
        });
        let lang_id = this.props.lang_id;
        let uuid=fetchUUid()
        let data = {
          firstname: this.state.firstname,
          lastname: this.state.lastname,
          email: this.state.emailRegister,
          country_code: this.state.country_code,
          mobile: this.state.mobile,
          password: this.state.password,
          locale: this.props.lang,
          uuid
        };

        axios
          .post(`${baseURL}user?lang_id=${lang_id}`, data, {
            headers: {
              // 'Content-Type': 'application/x-www-form-urlencoded'
            }
          })
          .then(res => {
            var response = res.data;

            if (response.status) {
              const {
                avatar,
                user_id,
                verified,
                mobile_verified,
                api_access_token
              } = response.data;

              this.props.dispatch(
                user.store_user_info({
                  avatar,
                  user_id,
                  verified,
                  mobile_verified
                })
              );

              GistToMember(this.props.lang, res.data.data.api_access_token);
            } else {
              this.setState({
                error_msg: response.message,
                is_loading: false
              });
            }
          })
          .catch(err => {
            this.setState({
              error_msg: err.response
                ? err.response.data
                  ? undefined !== err.response.data.error
                    ? err.response.data.error
                    : err.response.data.message
                  : ""
                : "",
              is_loading: false
            });
          });
      })
      .catch(formErrors => {});
    /*} else {
      this.setState({
        is_loading: false,
        is_busy: false,
        error_msg: this.props.locale.auth.robot_verify
      });
    }*/
  };

  verify_robot = value => {
    this.setState({
      verify_robot: true
    });
  };

  render() {
    if (this.state.loggedin) {
      return <Redirect from={"/register"} to={"/"} />;
    }

    return (
      <Page
        title={this.props.locale.auth.login.title}
        description={this.props.locale.auth.login.desc}
        {...this.props}
      >
        <div className="container">
          <div className="columns is-mobile">
            <div className="column is-3-desktop is-1-mobile is-2-tablet" />

            <div className="column is-6-desktop is-10-mobile is-8-tablet">
              <div className="form-wrap">
                <p className="page-title mar-bot-27">
                  {this.props.locale.auth.login.title}
                </p>

                <div className="columns is-mobile mar-bot-0">
                  <div className="column is-2-desktop is-hidden-mobile" />

                  <div className="column is-8-desktop is-full-mobile is-8-tablet">
                    <TButton
                      ariaLabel="Register with Facebook"
                      leftIcon={faFacebookSquare}
                      text={this.props.locale.auth.facebook_button_text}
                      className="facebook is-uppercase mar-bot-13 full"
                      onPress={() => loginFacebook(this.props.lang)}
                    />
                  </div>

                  <div className="column is-2-desktop is-hidden-mobile" />
                </div>

                <div className="is-flex or-separator">
                  <div className="line" />
                  <p>{this.props.locale.auth.seperator}</p>
                  <div className="line" />
                </div>

                <p className="form-helper mar-bot-15">
                  {this.props.locale.auth.form_helper_text}
                </p>

                {this.state.error_msg ? (
                  <div className="sys-msg has-background-danger mar-bot-15">
                    <FontAwesomeIcon icon={"exclamation-circle"} />
                    {this.state.error_msg}
                  </div>
                ) : null}

                <div className="columns is-mobile is-multiline">
                  <div className="column is-6-desktop is-full-touch">
                    <div className="columns is-mobile is-multiline">
                      <div className="column is-6-desktop is-6-touch">
                        <TextInput
                          required
                          placeholder={this.props.locale.auth.firstname}
                          id={"firstname-input"}
                          name="firstname"
                          type="text"
                          errorMessage={this.state.firstnameError}
                          onBlur={() =>
                            this.validation(["firstname"])
                              .then()
                              .catch(e => {})
                          }
                          onKeyPress={e => {
                            if (e.key === "Enter") {
                              this.register();
                            }
                          }}
                          onKeyUp={e => this.store_data(e, "firstname")}
                        />
                      </div>

                      <div className="column is-6-desktop is-6-touch">
                        <TextInput
                          required
                          placeholder={this.props.locale.auth.lastname}
                          type="text"
                          id={"lastname-input"}
                          name="lastname"
                          errorMessage={this.state.lastnameError}
                          onBlur={() =>
                            this.validation(["lastname"])
                              .then()
                              .catch(e => {})
                          }
                          onKeyPress={e => {
                            if (e.key === "Enter") {
                              this.register();
                            }
                          }}
                          onKeyUp={e => this.store_data(e, "lastname")}
                        />
                      </div>
                    </div>
                  </div>

                  <div className="column is-2-desktop is-2-tablet is-4-mobile">
                    <DropDown
                      ariaLabel="Select Country Code"
                      options={[{ value: "+2", title: "+2" }]}
                      onSelect={e => this.store_data(e, "country_code")}
                    />
                  </div>

                  <div className="column is-4-desktop is-4-tablet is-8-mobile">
                    <TextInput
                      required
                      placeholder={this.props.locale.auth.mobile}
                      type="number"
                      id={"mobile-input"}
                      name="mobile"
                      errorMessage={this.state.mobileError}
                      onBlur={() =>
                        this.validation(["mobile"])
                          .then()
                          .catch(e => {})
                      }
                      onKeyPress={e => {
                        if (e.key === "Enter") {
                          this.register();
                        }
                      }}
                      onKeyUp={e => this.store_data(e, "mobile")}
                    />
                  </div>
                </div>

                <div className="columns is-mobile">
                  <div className="column is-6">
                    <TextInput
                      required
                      placeholder={this.props.locale.auth.password_placeholder}
                      id={"password-input"}
                      name="password"
                      type="password"
                      errorMessage={this.state.passwordError}
                      onBlur={() => {
                        // if (this.state.password) {
                        this.validation(["password"])
                          .then()
                          .catch(e => {});
                        // }
                      }}
                      onKeyPress={e => {
                        if (e.key === "Enter") {
                          this.register();
                        }
                      }}
                      onKeyUp={e => this.store_data(e, "password")}
                    />
                  </div>

                  <div className="column is-6">
                    <TextInput
                      required
                      placeholder={this.props.locale.auth.email_placeholder}
                      type="email"
                      id={"email-input"}
                      name="emailRegister"
                      errorMessage={this.state.emailRegisterError}
                      onBlur={() => {
                        if (this.state.emailRegister) {
                          this.validation(["emailRegister"])
                            .then()
                            .catch(e => {});
                        }
                      }}
                      onKeyPress={e => {
                        if (e.key === "Enter") {
                          this.register();
                        }
                      }}
                      onKeyUp={e => this.store_data(e, "emailRegister")}
                    />
                  </div>
                </div>

                {/* <div className="mar-bot-15">
                  <ReCAPTCHA
                    sitekey={google_capcha}
                    onChange={this.verify_robot}
                  />
                </div> */}

                <TButton
                  ariaLabel="Register"
                  text={this.props.locale.auth.registration_button_text}
                  onPress={() => this.register()}
                  type="submit"
                  id={"register-button"}
                  name="register"
                  className={`button grad-blue mar-bot-25`}
                  loading={this.state.is_loading}
                />

                <div className="form-foot mar-auto has-text-centered">
                  {this.props.locale.auth.sign_in_text}{" "}
                  <a href={`/${this.props.lang}/login`}>
                    {this.props.locale.auth.sign_in_button}
                  </a>
                </div>
              </div>
            </div>

            <div className="column is-3-desktop is-1-mobile is-2-tablet" />
          </div>
        </div>
      </Page>
    );
  }
}
// export default Register;
export default connect(null)(Register);
