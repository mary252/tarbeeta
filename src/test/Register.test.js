import React from 'react';
import {shallow } from 'enzyme';
import {Register} from '../containers/auth/Register'
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import store from "../Store";
import * as user from '../actions/user'
import {baseURL} from '../common/Config'
import { GistToMember } from '../services'
import axios from "axios"; // v0.15.3
import httpAdapter from "axios/lib/adapters/http";
import nock from "nock";

configure({ adapter: new Adapter() });
jest.setTimeout(30000);

const match = {
    params : { 
                lang : 'en' //any id you want to set
             }
   }
  const location={
    search:''
  }

  const host = baseURL;

  axios.defaults.host = host;
  axios.defaults.adapter = httpAdapter;
  
describe('Register page test',()=>{
    const register = shallow(<Register  match={match} location={location}/>);

    it('container renders without errors', () => {
        expect(register.exists()).toEqual(true)
    })
    it("renders correctly", () => {
      expect(register).toMatchSnapshot();
    });
    let instance =register.instance()

    it('the locale load',()=>{
        expect(instance.state.locale).not.toBeNull()
    })

    it('store_data function test ',()=>{
        expect(instance.state.email).toEqual("")
        let event ={
            target:{
                value:'email'
            }
        }
        instance.store_data(event,'email')
        expect(instance.state.email).toEqual('email')
    })
    
    it('verify robot function test ',()=>{
        expect(instance.state. verify_robot).toBeFalsy()
        instance.verify_robot('12321')
        expect(instance.state. verify_robot).toBeTruthy()

    })

    it('register function test',async()=>{
        instance.setState({
            firstname: 'test',
            lastname: 'test',
            email: 'test@tarbeeta.com',
            country_code: '+2',
            mobile: '01027836541',
            password: '12345678',
            verify_robot: true
        })
        nock(host)
        .post("/user/")
        .reply(200, {
          data: {
            "access_token":'3242576786543567897654345678',
            avatar:'awaver',
            user_id:233,
            verified:1, 
            mobile_verified:1
          }
        });
        try{
            await instance.register()
            let access_token=localStorage.getItem('access_token')
            expect(access_token).toEqual('3242576786543567897654345678')
            let view=localStorage.gletItem('view_mode')
            expect(view).toEqual('buyer')
            expect(GistToMember).toBeCalled()
        }catch(e){

        }
    })

    it('first name input test',()=>{
      const firstname=register.find("#firstname-input")
      expect(firstname.length).toEqual(1);
      const event = { target: { name: "firstname", value: "456" } };
      firstname.simulate("keyUp", event);
      expect(instance.state.firstname).toEqual("456");
  
    })
    
    it('lastname input test',()=>{
      const lastname=register.find("#lastname-input")
  
      expect(lastname.length).toEqual(1);
      const event = { target: { name: "lastname", value: "456" } };
  
      lastname.simulate("keyUp", event);
  
      expect(instance.state.lastname).toEqual("456");
  
    })
    it('mobile input test',()=>{
      const mobile=register.find("#mobile-input")
      expect(mobile.length).toEqual(1);
      const event = { target: { name: "mobile", value: "456" } };
      mobile.simulate("keyUp", event);
      expect(instance.state.mobile).toEqual("456");
  
    })
    
    it('email input test',()=>{
      const username=register.find("#email-input")
      expect(username.length).toEqual(1);
      const event = { target: { name: "email", value: "456" } };
      username.simulate("keyUp", event);
      expect(instance.state.email).toEqual("456");
  
    })
    
    it('password input test',()=>{
      const password=register.find("#password-input")
  
      expect(password.length).toEqual(1);
      const event = { target: { name: "password", value: "456" } };
  
      password.simulate("keyUp", event);
  
      expect(instance.state.password).toEqual("456");
  
    })

    it ('register button working',()=>{
      const register_button=register.find("#register-button")
      const spy = jest.spyOn(instance, "register");
      register_button.prop("onPress")();
      expect(spy).toHaveBeenCalled()
    })
})

describe('redux dispatch ',()=>{
    let res=store.dispatch(
      user.store_user_info({
        avatar:null,
        user_id:1,
        verified:0,
        mobile_verified:1
      })
    );
    it('dispatch test',()=>{
      expect(res).toEqual({
        type: "STORE_USER_INFO",
        payload: {
          avatar:null,
          user_id:1,
          verified:0,
          mobile_verified:1
        }
      })
        
    })
    let user_state= store.getState().user
      it('state changed',()=>{
        expect(user_state.avatar).toBe(null)
        expect(user_state.verified).toBe(0)
        expect(user_state.mobile_verified).toBe(1)
      })
  })