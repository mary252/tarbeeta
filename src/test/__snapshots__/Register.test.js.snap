// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register page test renders correctly 1`] = `ShallowWrapper {}`;
